### PR TITLE
Rename content_metadata_creation to processing_configuration

### LINKED ISSUE #1180: Re-name "content metadata creation" to "processing configuration"
The Preassembly form contains a dropdown menu labeled "content metadata creation":

Preassembly content metadata creation menu

The label refers to "content metadata", which was the name given to structural metadata in the old Fedora model. The function of the menu is to provide an instruction for how Preassembly should create structural metadata. In other words, it is literally about "content metadata creation".

**Background: what the content metadata creation setting does**

Currently, there are two settings on this menu:

- "default" = every file will be made into its own "resource"
- "filename" = files will be grouped into "resources" based on filename patterns

The main use cases for the "filename" pattern is when the content being accessioned contains multiple files per unit of content: For example, an image or page could contain the following file types:

- TIFF image of page
- JP2 image of page
- PDF of page
- XML representation of OCR of page
- TXT representation of OCR of page

As long as each of those files shares the same filename - except for the extension - they will be grouped together. The image and book viewers expect this kind of grouping.

**Request Re-name that menu to something that makes more sense to users**

Since there's no such thing as "content metadata" anymore, we should change the name. "Structural metadata creation" sounds too literal. Since the menu is about how files should be processed, we could go with a more generic name like: "Processing configuration".

Then we could rename the options on the menu:

- "Default" --> leave this alone
- "Filename" --> "Group by filename"

The result will be:

- Menu title: "Processing configuration"
- Menu options:
 - "Default"
 - "Group by filename"

I want to add another option beyond those two but I will file a new ticket for that.

### PR DESCRIPTION
**Pull request by peetucket**
# Why was this change made? 🤔

Fixes #1180 - rename "content_metadata_creation" to "processing_configuration" in the UI and in the code, including renaming of a database column and all related methods and attributes.  Also change the display value of one of the possible values (but not the underlying value stored in the database, so changes needed there).

Given the large number of code changes, need to validate this on stage by:

- [x] run pre-assembly integration test - required changes: https://github.com/sul-dlss/infrastructure-integration-test/pull/614
- [x] have Andrew check it out

# How was this change tested? 🤨

Updated specs, integration tests

### CODE REVIEW

File: app/lib/pre_assembly/batch.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Batch
                   :objects_had_errors
 
     delegate :staging_location,
-             :content_md_creation,
+             :processing_configuration,
              :content_structure,
```
Comment (peetucket) [inline app/lib/pre_assembly/batch.rb:19]: note: `content_md_creation` == `content_metadata_creation` in `BatchContext` ... so we can rename the delegation and then get rid of that level of indirection in the class we are delegating to

File: app/lib/pre_assembly/digital_object.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class DigitalObject
                 :stager, :label, :druid, :source_id, :container
 
     delegate :staging_location,
-             :content_md_creation,
+             :processing_configuration,
              :content_structure,
```
Comment (peetucket) [inline app/lib/pre_assembly/digital_object.rb:13]: note: `content_md_creation` is now `processing_configuration` in `Batch` as noted above

File: app/lib/pre_assembly/digital_object.rb [hunk 3]
```diff
@@ -142,14 +142,14 @@ def build_structural
                                          reading_order:)
       else
         build_from_staging_location(objects: object_files.sort,
-                                    content_metadata_creation: content_md_creation.to_sym,
+                                    processing_configuration:,
                                     content_md_creation_style:,
```
Comment (peetucket) [inline app/lib/pre_assembly/digital_object.rb:146]: unnecessary indirection of names removed, and `to_sym` done in the class initialize itself now)

File: app/models/batch_context.rb [hunk 4]
```diff
@@ -63,11 +62,6 @@ def file_manifest
                                   staging_location:)
   end
 
-  # this method would be better named manifest_type, but we are using the name content_metadata_creation in the UI
-  def content_md_creation
-    content_metadata_creation
-  end
-
```
Comment (peetucket) [inline app/models/batch_context.rb:70]: unnecessary indirection -- just reference our new name of `processing_configuration` everywhere it is needed

File: app/services/discovery_report.rb [hunk 5]
```diff
@@ -8,7 +8,7 @@ class DiscoveryReport
   attr_reader :batch, :summary
   attr_accessor :error_message, :objects_had_errors
 
-  delegate :staging_location, :content_md_creation, :project_style, :file_manifest, to: :batch
+  delegate :staging_location, :project_style, :file_manifest, to: :batch
```
Comment (peetucket) [inline app/services/discovery_report.rb:12]: `content_md_creation` does not appear to be used in the discovery report anyway

**Review by ndushay — APPROVED** [review]
I'm approving this but would like Andrew to sign off on it, perhaps by having him see it deployed in stage or qa and exercising it???

**Comment by peetucket** [comment]
> I'm approving this but would like Andrew to sign off on it, perhaps by having him see it deployed in stage or qa and exercising it???

I asked him about this after deploying to stage and he said he would not do anything more comprehensive than the integration tests already do.  So I took that as sign off.

**Comment by andrewjbtw** [comment]
Yes - although if the integration test don't use the "filename" grouping option, I should check that specifically. Come to think of it, if they only use a single image then they don't test this specifically.

**Comment by peetucket** [comment]
The `spec/features/preassembly_image_accessioning_spec.rb` integration test does select the "Group by filename" option but I am not 100% sure our expectations verify it actually groups the files as expected, so perhaps it is worth a double check.

**Comment by andrewjbtw** [comment]
Ok, I've tested both configuration options with some image/book content. It all looks good to me.

**Pull request merged**